server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -19,11 +19,41 @@ def __init__(self, data, graph_method="umap", diffexp_method="ttest"):
         self._add_mandatory_annotations()
         self.cell_count = self.data.shape[0]
         self.gene_count = self.data.shape[1]
+        self._create_schema()
         self.graph_method = graph_method
         self.diffexp_method = diffexp_method
 
-    def _set_cell_names(self):
-        self.data.obs["cell_name"] = list(self.data.obs.index)
+    def _create_schema(self):
+        self.schema = {
+            "dataframe": {
+                "nObs": self.cell_count,
+                "nVar": self.gene_count,
+                "type": str(self.data.X.dtype)
+            },
+            "annotations": {
+                "obs": [],
+                "var": []
+            }
+        }
+        for ax in Axis:
+            curr_axis = getattr(self.data, str(ax))
+            for ann in curr_axis:
+                ann_schema = {"name": ann}
+                data_kind = curr_axis[ann].dtype.kind
+                if data_kind == 'f':
+                    ann_schema["type"] = "float32"
+                elif data_kind in ['i', 'u']:
```

Comment:
The current API only supports in32.   'i' and 'u' could be any of int32, int64, uint32 and uint64, three of which will potentially overflow an int32.

Reply:
I was worried about float16, int16, and int8. I wasn't sure if this was the place to issue warnings about data type yet.